### DISCUSSION TITLE
How does the MS1 annotation work?

### DISCUSSION THREAD

**Discussion by Adafede** [taxonomicallyinformedannotation/tima#128] [Q&A]
Original question posted by @adrien-jagora in #26:

> I had one additionnal question regarding the optional MS1-based annotation on Lotus. If I understand correctly, this happens when the score S1 equals 0 as a way to allow TimaR to do its taxonomical annotation later on. Is this MS1 dereplication based on the exact mass of the precursor ion considered or on a molecular formula previously generated ? Thanks in advance !

**Comment by Adafede** [comment]
**Answer**:

The MS1 annotation tries to annotate all features, independently of their S1 score. If a feature has an MS2 annotation, the MS1 annotation part will still be performed, but its result will have a spectral score of 0 (logically).

This allows to compensate for multiple shortcomings at once:
- Annotate features without fragmentation spectrum
- Annotate features with fragmentation spectrum but no library match, based on exact mass match
- Annotate weird, rarely occurring adducts/losses/dimers/multicharged

The MS1 annotation is performed as follows:

Depending on the ionization mode, `[M+H]+` and `[M-H]-` are always considered. Then, within a relatively small retention time window, the differences between the precursor masses of all features within that window are computed.
_Example_:
You have 10 features between 1.25 and 1.30 min, it calculated the mass difference between all of them and then tries to match it to list of known _adducts_,_clusters_, or _in source fragments or dimers_.

The underlying lists are available at [inst/extdata](https://github.com/taxonomicallyinformedannotation/tima-r/blob/main/inst/extdata/).

These are pre-combined with lists of known masses/molecular formulas of interest, by default the ones in [inst/params/default/prepare_libraries_sop_merged.yaml](https://github.com/taxonomicallyinformedannotation/tima-r/blob/main/inst/params/default/prepare_libraries_sop_merged.yaml) and [inst/params/default/prepare_libraries_adducts.yaml](https://github.com/taxonomicallyinformedannotation/tima-r/blob/main/inst/params/default/prepare_libraries_adducts.yaml).

You can see the default parameters for the MS1 annotation at [inst/params/default/annotate_masses.yaml](https://github.com/taxonomicallyinformedannotation/tima-r/blob/main/inst/params/default/annotate_masses.yaml)



I hope this clarifies. If not, do not hesitate and I'll try developing further. 😉

**Reply by adrien-jagora** [reply]
Well, I think I have all the information I need ! Thank you very much Adriano !
